llvm/lib/Target/AMDGPU/AMDGPUAtomicOptimizer.cpp

```diff
@@ -178,6 +178,20 @@ bool AMDGPUAtomicOptimizerImpl::run(Function &F) {
   return Changed;
 }
 
+static bool shouldOptimizeForType(Type *Ty) {
+  switch (Ty->getTypeID()) {
+  case Type::FloatTyID:
+  case Type::DoubleTyID:
+    return true;
+  case Type::IntegerTyID: {
+    if (Ty->getIntegerBitWidth() == 32 || Ty->getIntegerBitWidth() == 64)
+      return true;
+  default:
```

Comment:
Don't forget pointers. In a follow up the should really just handle half / bfloat and vectors

Reply:
I feel pointers should be handled as a follow up too since I intend this patch to reflect current requirements (changed the title since it was misleading)

Reply:
Also enabling for half, bfloat etc would require additional legalization support for intrinsics such as update.dpp , set.incactive.lane ....

Reply:
I thought those all supported 16-bit values already